src/xo-app/vm/tab-network.js

```diff
@@ -149,159 +145,226 @@ class VifItem extends BaseComponent {
+          <XoSelect
+            onChange={_setNetwork}
+            predicate={_getNetworkPredicate()}
+            value={network}
+            xoType='network'
+          >
+            {network.name_label}
+          </XoSelect>
+        )
+      )
+    }),
+    name: _('vifNetworkLabel'),
+  },
+  {
+    component: VifAllowedIps,
+    name: _('vifAllowedIps'),
+  },
+  {
+    component: connectStore(() => {
+      const getVif = (state, props) => props.item
+      const getNetworkId = createSelector(getVif, vif => vif.$network)
+      const getNetwork = createGetObject(getNetworkId)
+      return (state, props) => ({
+        vif: getVif(state, props),
+        network: getNetwork(state, props),
+        networkId: getNetworkId(state, props),
+      })
+    })(({ vif, network }) => {
+      const _getIps = createSelector(
+        () => vif.allowedIpv4Addresses || EMPTY_ARRAY,
+        () => vif.allowedIpv6Addresses || EMPTY_ARRAY,
+        concat
+      )
+      const _getNetworkStatus = () => {
+        if (!isEmpty(_getIps())) {
+          return (
+            <Tooltip content={_('vifLockedNetwork')}>
+              <Icon icon='lock' />
+            </Tooltip>
+          )
+        }
+        if (!network) {
+          return (
+            <Tooltip content={_('vifUnknownNetwork')}>
+              <Icon icon='unknown-status' />
+            </Tooltip>
+          )
+        }
+        if (network.defaultIsLocked) {
+          return (
+            <Tooltip content={_('vifLockedNetworkNoIps')}>
+              <Icon icon='error' />
+            </Tooltip>
+          )
+        }
+        return (
+          <Tooltip content={_('vifUnLockedNetwork')}>
+            <Icon icon='unlock' />
+          </Tooltip>
+        )
+      }
+
+      return (
+        <div>
           <StateButton
             disabledLabel={_('vifStatusDisconnected')}
-            disabledHandler={isVmRunning && connectVif}
+            disabledHandler={isVmRunning ? connectVif : noop}
             disabledTooltip={_('vifConnect')}
             enabledLabel={_('vifStatusConnected')}
             enabledHandler={disconnectVif}
             enabledTooltip={_('vifDisconnect')}
             handlerParam={vif}
             state={vif.attached}
           />{' '}
-          {this._getNetworkStatus()}
-        </td>
-        <td className='text-xs-right'>
-          {!vif.attached && (
-            <ActionRowButton
-              handler={deleteVif}
-              handlerParam={vif}
-              icon='remove'
-              tooltip={_('vifRemove')}
-            />
-          )}
-        </td>
-      </tr>
-    )
-  }
+          {_getNetworkStatus()}
+        </div>
+      )
+    }),
+    name: _('vifStatusLabel'),
+  },
+]
+const GROUPED_ACTIONS = [
+  {
+    disabled: selectedItems => {
+      const vifsAttached = filter(selectedItems, vif => {
+        if (vif.attached) return true
+      })
+      return selectedItems.length === vifsAttached.length
```

Comment:
This whole test would be simpler with [`_.some`](http://devdocs.io/lodash~4/index#some).

```js
{
  disabled: selectedItems => some(selectedItems, 'attached'),
  ...
}
```